var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -0,0 +1,118 @@
+    variant('mpi', default=True,
+            description='Enable MPI')
+    # transforms (not yet implemented)
+    # variant('bzip2', default=True,
+    #         description='Enable BZip2 compression')
+    # variant('zfp', default=True,
+    #         description='Enable ZFP compression')
+    # transport engines
+    variant('dataman', default=True,
+            description='Enable the DataMan engine for WAN transports')
+    # currently required by DataMan, optional in the future
+    # variant('zeromq', default=False,
+    #         description='Enable ZeroMQ for the DataMan engine')
+    variant('hdf5', default=False,
+            description='Enable the HDF5 engine')
+    variant('adios1', default=False,
+            description='Enable the ADIOS 1.x engine')
+    # language bindings
+    variant('python', default=True,
+            description='Enable the Python >= 2.7 bindings')
+
+    # requires mature C++11 implementations
+    conflicts('%gcc@:4.7')
+    conflicts('%intel@:15')
+    conflicts('%pgi@:14')
+
+    # DataMan needs dlopen
+    conflicts('+dataman', when='~shared')
+
+    depends_on('cmake@3.5.0:', type='build')
+
+    # contained in thirdparty/
+    # depends_on('googletest')
+    # depends_on('pugixml')
+    # depends_on('kwsys')
+    # depends_on('nlohmannjson')
+    # depends_on('pybind11@2.1.1:', when='+python')
+
+    depends_on('mpi', when='+mpi')
+
+    depends_on('hdf5', when='+hdf5')
+    depends_on('hdf5+mpi', when='+hdf5+mpi')
+    depends_on('adios', when='+adios1')
+    depends_on('adios+mpi', when='+adios1+mpi')
+
+    depends_on('bzip2', when='+bzip2')
+    depends_on('zfp', when='+zfp')
+
+    extends('python', when='+python')
+    depends_on('python@2.7:', when='+python')
+    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='+python')
+    depends_on('py-mpi4py@2.0.0:', type=('build', 'run'), when='+mpi +python')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            '-DADIOS2_BUILD_SHARED_LIBS:BOOL={0}'.format((
+                'ON' if '+shared' in spec else 'OFF')),
+            '-DADIOS2_BUILD_TESTING=OFF',
+            '-DADIOS2_USE_MPI={0}'.format((
+                'ON' if '+mpi' in spec else 'OFF')),
+            '-DADIOS2_USE_BZip2={0}'.format((
+                'ON' if '+bzip2' in spec else 'OFF')),
+            '-DADIOS2_USE_ZFP={0}'.format((
+                'ON' if '+zfp' in spec else 'OFF')),
+            '-DADIOS2_USE_DataMan={0}'.format((
+                'ON' if '+dataman' in spec else 'OFF')),
+            '-DADIOS2_USE_ZeroMQ={0}'.format((
+                'ON' if '+dataman' in spec else 'OFF')),
+            '-DADIOS2_USE_HDF5={0}'.format((
+                'ON' if '+hdf5' in spec else 'OFF')),
+            '-DADIOS2_USE_ADIOS1={0}'.format((
+                'ON' if '+adios1' in spec else 'OFF')),
+            '-DADIOS2_USE_Python={0}'.format((
+                'ON' if '+python' in spec else 'OFF'))
```

Comment:
A couple things to watch out for:

1. Does it use the correct Python? Some CMake projects have a flag that lets you pick the correct Python executable to use. It should be first in your `PATH` though.
2. Does it install the Python libraries to the correct location (`<prefix>/lib/python2.7/site-packages`, where `<prefix>` is the prefix of Adios2 and `2.7` is the MAJOR.MINOR version of Python)? I've seen too many packages that installed their Python libraries to the Python installation prefix instead of their own prefix. Or they installed them to a different directory hierarchy than `lib/python2.7/site-packages`. Some CMake packages have an option to control this.